src/Services/Catalog/Catalog.API/01-Services/ICatalogService.cs

```diff
@@ -1,54 +1,93 @@
-    }
+    IEnumerable<CatalogCategoryDTO> GetAllCategories();
+    CatalogCategoryDTO GetCategoryById(int id);
+    CatalogItemDataResult GetAllItems(CatalogItemFilter filter);
+}
+
+public class CatalogService(ApplicationDataContext context) : ICatalogService
+{
+    private readonly ApplicationDataContext _context = context;
 
-    public class CatalogService(ApplicationDataContext context) : ICatalogService
+    public IEnumerable<CatalogCategoryDTO> GetAllCategories()
     {
-        private readonly ApplicationDataContext _context = context;
-
-        public CatalogItemDataResult GetAll(CatalogItemFilter filter)
-        {
-            filter ??= new CatalogItemFilter() { PageIndex = 0, PageSize = 10, ShowOnlyHighlighted = true };
-
-            var totalItems = _context.CatalogItems.Count(w => w.IsHighlighted == filter.ShowOnlyHighlighted);
-
-            var data = _context.CatalogItems
-                .Where(w => w.IsHighlighted == filter.ShowOnlyHighlighted)
-                .Skip(filter.PageIndex * filter.PageSize)
-                .Take(filter.PageSize)
-                .AsNoTracking()
-                .Select(item => new CatalogItemDTO
-                {
-                    Id = item.Id,
-                    Name = item.Name,
-                    Description = item.Description,
-                    Brand = item.Brand.Name,
-                    Price = item.Price
-                });
-
-            return new CatalogItemDataResult() { TotalItems = totalItems, Items = data };
-        }
+        return _context.CatalogCategories.Select(c => new CatalogCategoryDTO { Id = c.Id, Name = c.Name });
     }
 
-    public class CatalogItemFilter
+    public CatalogCategoryDTO GetCategoryById(int id)
     {
-        public bool ShowOnlyHighlighted { get; set; }
-        public int PageIndex { get; set; }
-        public int PageSize { get; set; }
+        var category = _context.CatalogCategories.FirstOrDefault(c => c.Id == id);
+
+        if (category is null)
+            return null;
+
+        return new CatalogCategoryDTO { Id = category.Id, Name = category.Name };
     }
 
-    public class CatalogItemDataResult
+    public CatalogItemDataResult GetAllItems(CatalogItemFilter filter)
     {
-        public int TotalItems { get; set; }
-        public IEnumerable<CatalogItemDTO> Items { get; set; }
+        filter ??= new CatalogItemFilter();
+
+        if (filter.PageSize <= 0)
+            filter.PageSize = 10;
+
+        if (filter.PageSize > 50)
+            filter.PageSize = 50;
+
+        var query = _context.CatalogItems.AsQueryable();
+
+        if (filter.ShowOnlyHighlighted) // Aqui o comportamento foi mudado,
+                                        // antes só mostrava produtos com destaque ou sem, eles não mostravam-se juntos.
+            query = query.Where(w => w.IsHighlighted);
+
+        var category = GetCategoryById(filter.CategoryId);
+
+        if (category is not null)
```

Comment:
aqui pode ser unificado para não ter diversos ifs.
.Where(w => category is null || w.Category.Id == category.Id)

Quase certeza que isso funciona como o if()